nntrainer/tensor/float_tensor.h

```diff
@@ -0,0 +1,1830 @@
+/**
+ * Copyright (C) 2023 Samsung Electronics Co., Ltd. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ * @file	float_tensor.h
+ * @date	09 November 2023
+ * @brief	This is FloatTensor class for 32-bit floating point calculation
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Donghyeon Jeong <dhyeon.jeong@samsung.com>
+ * @bug		No known bugs except for NYI items
+ */
+
+#ifndef __FLOAT_TENSOR_H__
+#define __FLOAT_TENSOR_H__
+#ifdef __cplusplus
+
+#include <array>
+#include <functional>
+#include <memory>
+#include <random>
+#include <stdexcept>
+#include <vector>
+
+#include <blas_interface.h>
+#include <iostream>
+#include <memory_data.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <tensor_dim.h>
+#include <util_func.h>
+
+#ifdef DEBUG
+#define EXCEPT_WHEN_DEBUG
+#else
+#define EXCEPT_WHEN_DEBUG noexcept
+#endif
+
+#define MAKE_SHARED_FLOAT_TENSOR(...) \
+  std::make_shared<nntrainer::FloatTensor>(__VA_ARGS__)
+
+#define CREATE_IF_EMPTY_DIMS_FLOAT(tensor, ...) \
+  do {                                          \
+    if (tensor.empty())                         \
+      tensor = FloatTensor(__VA_ARGS__);        \
+  } while (0);
+
+namespace nntrainer {
+
+using TensorDim = ml::train::TensorDim;
+using Tformat = ml::train::TensorDim::Format;
+using Tdatatype = ml::train::TensorDim::DataType;
+
+class LazyTensor;
+class SrcSharedTensor;
+
+/**
+ * @class   FloatTensor Class for Calculation
+ * @brief   FloatTensor Class for Calculation
+ */
+class FloatTensor {
+public:
+  /**
+   * @brief     Enumeration of Weight Initialization Type
+   * @todo      support intialization from file
+   */
+  enum class Initializer {
```

Comment:
this Initializer enum can be shared to other tensors.

Reply:
I will replace it with `TensorV2::Initializer` with the upcoming change